packages/gatsby/src/query/index.ts

```diff
@@ -211,7 +212,7 @@ export async function processStaticQueries(
   queryIds: IGroupedQueryIds["staticQueryIds"],
   { state, activity, graphqlRunner, graphqlTracing }
 ): Promise<void> {
-  return processQueries<string>({
+  const processedQueries = processQueries<string>({
```

Comment:
I think we need to await on that before flushing?
```suggestion
  const processedQueries = await processQueries<string>({
```